app/src/main/java/com/getsensibill/spendmanager/demo/capture_flow/CaptureFlowJavaActivity.java

```diff
@@ -0,0 +1,84 @@
+ *
+ * This is an example on how you can track the state changing during a capture flow of a receipt
+ * using the [CaptureFlowCoordinator].
+ */
+public class CaptureFlowJavaActivity extends AppCompatActivity {
+    private ActivityCaptureFlowJavaBinding binding;
+
+    // Creates a capture flow Coordinator
+    private CaptureFlowCoordinator captureFlow = new CaptureFlowCoordinator(this);
+
+    // Setup the capture flow listener which is used to listen to the state changing
+    private CaptureFlowCoordinator.CaptureFlowListener captureFlowListener =
+            new CaptureFlowCoordinator.CaptureFlowListener() {
+        @Override
+        public void onCaptureFlowUpdate(@NotNull CaptureFlowState newState,
+                                        @Nullable String externalAccountTransactionId) {
+            String text = "\n " + new Date().toString() + ": ";
+
+            if (newState instanceof CaptureFlowState.IMAGES_CAPTURED) {
+                // Handle the state that images have just been captured
+                text += "Images are captured";
+
+            } else if (newState instanceof CaptureFlowState.FLOW_CANCELLED) {
+                // Handle the user cancelling the flow
+                text += "Capture flow cancelled";
+
+            } else if (newState instanceof CaptureFlowState.Error) {
+                // Handle an error occurring during the flow
+                final CaptureFlowState.Error errorState = (CaptureFlowState.Error) newState;
+                final Exception error = errorState.getException();
+                text += "Error occurred: " + error.getMessage();
+
+            } else if (newState instanceof  CaptureFlowState.Transacting) {
+                // Handle receipt upload transaction updates that will occur after the image has been captured
+                // and the receipt is uploading
+                final CaptureFlowState.Transacting transactingState = ((CaptureFlowState.Transacting) newState);
+                final Transaction transaction = transactingState.getTransaction();
+
+                text += "Transacting";
+                text += "\nstatus: " + transaction.getStatus().toString();
+                text += "\nnlocalId: " + transaction.getLocalId();
+                text += "\nntxnId: " + transaction.getTransactionId();
+                text += "\nnreceiptId: " + transaction.getReceiptId();
+                text += "\n(savedExtTxnId: " + externalAccountTransactionId + ")";
+            }
+
+            String finalText = text;
+            runOnUiThread(() -> {
+                binding.progressText.append(finalText);
+            });
+        }
+    };
+
+    @Override
+    protected void onCreate(@androidx.annotation.Nullable @Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityCaptureFlowJavaBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        binding.launchCapture.setOnClickListener(view ->
```

Comment:
do you think we should change this to classic java `new View.OnClickListener...`

Reply:
AS suggested it. Otherwise it gives warnings about not using the lambda